localstack/services/secretsmanager/provider.py

```diff
@@ -0,0 +1,357 @@
+import json
+import logging
+import random
+import string
+from abc import ABC
+from typing import Optional
+
+from moto.iam.policy_validation import IAMPolicyDocumentValidator
+from moto.secretsmanager import models as secretsmanager_models
+from moto.secretsmanager.exceptions import SecretNotFoundException
+from moto.secretsmanager.models import SecretsManagerBackend, secretsmanager_backends
+from moto.secretsmanager.responses import SecretsManagerResponse
+
+from localstack.aws.api import RequestContext, ServiceResponse
+from localstack.aws.api.secretsmanager import (
+    AddReplicaRegionListType,
+    BooleanType,
+    CancelRotateSecretResponse,
+    ClientRequestTokenType,
+    CreateSecretResponse,
+    DeleteResourcePolicyResponse,
+    DeleteSecretResponse,
+    DescribeSecretResponse,
+    DescriptionType,
+    GetResourcePolicyResponse,
+    GetSecretValueResponse,
+    KmsKeyIdType,
+    ListSecretVersionIdsResponse,
+    MaxResultsType,
+    NameType,
+    NextTokenType,
+    NonEmptyResourcePolicyType,
+    PutResourcePolicyResponse,
+    PutSecretValueResponse,
+    RecoveryWindowInDaysType,
+    RemoveRegionsFromReplicationResponse,
+    RemoveReplicaRegionListType,
+    ReplicateSecretToRegionsResponse,
+    RestoreSecretResponse,
+    RotateSecretResponse,
+    RotationLambdaARNType,
+    RotationRulesType,
+    SecretBinaryType,
+    SecretIdType,
+    SecretsmanagerApi,
+    SecretStringType,
+    SecretVersionIdType,
+    SecretVersionStagesType,
+    SecretVersionStageType,
+    StopReplicationToReplicaResponse,
+    TagKeyListType,
+    TagListType,
+    UpdateSecretResponse,
+    UpdateSecretVersionStageResponse,
+    ValidateResourcePolicyResponse,
+)
+from localstack.constants import TEST_AWS_ACCOUNT_ID
+from localstack.services.moto import call_moto, call_moto_with_request
+from localstack.utils.aws import aws_stack
+from localstack.utils.common import to_str
+
+LOG = logging.getLogger(__name__)
+
+
+class SecretsmanagerProvider(SecretsmanagerApi, ABC):
+    def __init__(self):
+        super().__init__()
+        apply_patches()
+
+    @staticmethod
+    def _transform_context_secret_id(context: RequestContext) -> Optional[dict]:
```

Comment:
nit: to indicate a dict type hint we don't use the `dict` constructor, instead we use `typing.Dict` -> `Optional[Dict]`